aeternity/__main__.py

```diff
@@ -0,0 +1,139 @@
+import sys
+
+from aeternity import EpochClient, Name, InvalidName
+
+
+def print_usage():
+    print('''aeternity cli tool:
+Usage:
+    aens available <domain.aet>
+            Check Domain availablity    
+    aens register <domain.aet> [--no-input]
+            Register a domain (incurs fees!)    
+    aens status <domain.aet>
+            Checks the status of a domain    
+    aens update <domain.aet> <address>
+            Updates where the name points to    
+    aens revoke <domain.aet> [--no-input]
+            Removes this domain from the block chain (incurs fees!)    
+    aens transfer <domain.aet> <receipient_address> [--no-input]
+            Transfers a domain to another user
+The `--no-input` will suppress any questions before performing the action.
+
+''')
+    sys.exit(1)
+
+def stderr(*args):
+    for message in args:
+        sys.stderr.write(message)
+        sys.stderr.write(' ')
+    sys.stderr.write('\n')
+    sys.stderr.flush()
+
+
+def prompt(message, skip):
+    if skip:
+        return True
+    if input(message + ' [y/N]') != 'y':
+        print('cancelled')
+        sys.exit(0)
+
+
+args = sys.argv[1:]
+if len(args) < 2:
+    print_usage()
+
+client = EpochClient()
+
+system, command = args[:2]
+
+# allow using the --no-input parameter anywhere
+noinput = False
+if '--no-input' in args:
+    args.remove('--no-input')
+    noinput = True
+
+if system == 'aens':
+    domain = args[2]
+    try:
+        Name.validate_name(domain)
+    except InvalidName as exc:
+        stderr('Error:', str(exc))
+        sys.exit(1)
+
+    name = Name(domain)
+
+    if command == 'available':
+        if name.is_available():
+            print(f'{domain} is available')
+            sys.exit(0)
+        else:
+            print(f'{domain} is not available')
+            sys.exit(1)
+
+    if command == 'register':
+        if not name.is_available():
+            print('Name was already claimed')
+            sys.exit(1)
+        prompt('Do you want to register this name? (incurs fees)', skip=noinput)
+        print('Name is available, pre-claiming now')
+        name.preclaim()
+        print('Pre-Claim successful')
+        print('Claiming name (waiting for next block, this may take a while)')
+        name.claim_blocking()
```

Comment:
But it's not safe per se. The claiming might fail if someone else preclaimed the name before you in the last block.

So I think the catch is really that it takes ages for this method to return, because it's not just a quick lookup in the database but a really log and blocking call (alas the name).

If you want to return directly from the call you can use the non-blocking variant `claim` that will raise an Exception if it's still to early to claim the name. This is similar to the node people prefixing their calls with `sync_` because all their code tends to be async and they want to signal that something is different.